edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/authentication/AuthAgentController.cs

```diff
@@ -57,7 +59,8 @@ await this.GetIdsFromUsername(request)
                                     async creds => (await this.AuthenticateAsync(creds), Option.Some(creds)),
                                     () => Task.FromResult((false, Option.None<IClientCredentials>())));
 
-                return this.Json(GetAuthResult(isAuthenticated, credentials));
+                var authResult = await GetAuthResult(isAuthenticated, credentials, this.metadataStore);
```

Comment:
Would not be better to create a new StoreMetadata(credentials, this.metadataStore) method, and call that separately (if isAuthenticated == true), so then no need to change the GetAuthResult()? I feel it should not be the concern of GetAuthResult() to do other things than create the object that represents the result.

Reply:
Good call. I'll make the change.